src/med_bench/utils/classifiers.py

```diff
@@ -0,0 +1,423 @@
+        x_clf = classifier_x.fit(x[train_index, :], t[train_index])
+        xm_clf = classifier_xm.fit(np.hstack((x, m))[train_index, :], t[train_index])
+        p_x[test_index] = x_clf.predict_proba(x[test_index, :])[:, 1]
+        p_xm[test_index] = xm_clf.predict_proba(np.hstack((x, m))[test_index, :])[:, 1]
+
+    return p_x, p_xm
+
+def get_y_m_classifiers(regularization, forest, calibration, calib_method):
+    if regularization:
+        alphas = ALPHAS
+        cs = ALPHAS
+    else:
+        alphas = [TINY]
+        cs = [np.inf]
+
+    if not forest:
+        clf_y = RidgeCV(alphas=alphas, cv=CV_FOLDS)
+        clf_m = LogisticRegressionCV(Cs=cs, cv=CV_FOLDS)
+    else:
+        clf_y = RandomForestRegressor(n_estimators=100, min_samples_leaf=10)
+        clf_m = RandomForestClassifier(n_estimators=100, min_samples_leaf=10)
+    if calibration:
+        clf_m = CalibratedClassifierCV(clf_m, method=calib_method)
+
+    return clf_y, clf_m
+
+
+def estimate_f_mu(t, m, x, y, crossfit, classifier_y, classifier_m, interaction):
+
+
+    n = len(y)
+    if len(x.shape) == 1:
+        x = x.reshape(-1, 1)
+    if len(m.shape) == 1:
+        mr = m.reshape(-1, 1)
+    else:
+        mr = np.copy(m)
+    if len(t.shape) == 1:
+        t = t.reshape(-1, 1)
+
+    t0 = np.zeros((n, 1))
+    t1 = np.ones((n, 1))
+    m0 = np.zeros((n, 1))
+    m1 = np.ones((n, 1))
+
+    train_test_list = get_train_test_lists(crossfit, n, x)
+
+    mu_11x, mu_10x, mu_01x, mu_00x = [np.zeros(n) for _ in range(4)]
+    f_00x, f_01x, f_10x, f_11x = [np.zeros(n) for _ in range(4)]
+
+    x_t_mr = get_interactions(interaction, x, t, mr)
+    t_x = get_interactions(interaction, t, x)
+
+    x_t1_m1 = get_interactions(interaction, x, t1, m1)
+    x_t1_m0 = get_interactions(interaction, x, t1, m0)
+    x_t0_m1 = get_interactions(interaction, x, t0, m1)
+    x_t0_m0 = get_interactions(interaction, x, t0, m0)
+
+    t0_x = get_interactions(interaction, t0, x)
+    t1_x = get_interactions(interaction, t1, x)
+
+    for train_index, test_index in train_test_list:
+
+        clf_y = classifier_y.fit(x_t_mr[train_index, :], y[train_index])
+        clf_m = classifier_m.fit(t_x[train_index, :], m.ravel()[train_index])
+        mu_11x[test_index] = clf_y.predict(x_t1_m1[test_index, :])
+        mu_10x[test_index] = clf_y.predict(x_t1_m0[test_index, :])
+        mu_01x[test_index] = clf_y.predict(x_t0_m1[test_index, :])
+        mu_00x[test_index] = clf_y.predict(x_t0_m0[test_index, :])
+        f_00x[test_index] = clf_m.predict_proba(t0_x[test_index, :])[:, 0]
+        f_01x[test_index] = clf_m.predict_proba(t0_x[test_index, :])[:, 1]
+        f_10x[test_index] = clf_m.predict_proba(t1_x[test_index, :])[:, 0]
+        f_11x[test_index] = clf_m.predict_proba(t1_x[test_index, :])[:, 1]
+
+    f = f_00x, f_01x, f_10x, f_11x
+    mu = mu_11x, mu_10x, mu_01x, mu_00x
+    return f, mu
+
+def get_y_m_x_classifiers(regularization, forest, calibration, calib_method):
```

Comment:
Same here regarding docstring

Reply:
This has been addressed in 5c5d7c0b812d2aa64f6ae01e519138b341c2f8a5